### DISCUSSION TITLE
How to fix your config.json

### DISCUSSION THREAD

**Discussion by NiceGuyIT** [Ylianst/MeshCentral#5004] [General]
A common problem in the issues is an incorrect `config.json`. What makes a config incorrect? How can you verify your config is correct?

Easy! Use [Visual Studio Code][vscode] to edit your `config.json` and add the schema at the top.

1. If you haven't already, download VS [code][vscode].
2. Download or copy the `config.json` to your computer.
3. Open `config.json` in `code` and add the schema as the top line. This schema is the raw JSON file in the MeshCentral repo.

```json
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "settings": {
    "your settings go here": "..."
  }
}
```

Now you have autocomplete, auto-format and validation for your `config.json`! If you start typing, Code will show the values that are valid for the location you are editing. Words with a red squiggle line are errors. Words with a orange squiggle line are warnings. Hover over both to see the error message and possible fixes. Code can even [format your config][].

While this is a huge step up, it's not perfect. If you notice, there are some invalid keys in the screenshot. This is perfectly valid JSON and MeshCentral will ignore them (maybe?). If you paste some configs into the wrong section, `code` will not tell you it's in the wrong section. Autocomplete will tell you what keys are valid and the type of the value (i.e. string, number, boolean).

Hopefully this will help verify your config is syntactically correct and prevent needless formatting errors, misspellings, etc.

![image](https://user-images.githubusercontent.com/7763429/221193591-78b2536e-7fa7-4d02-9663-1cbd46ed8f12.png)

[vscode]: https://code.visualstudio.com/download

[format your config]: https://stackoverflow.com/questions/29973357/how-do-you-format-code-in-visual-studio-code-vscode

**Comment by PrplHaz4** [comment]
It's important to note the schema has had lots of additions over the last couple years, so double-check your schema points to the right place (if you copied it from the website years ago like I did).  

Anything pointing to the schema on the old website will be a good bit out of date by now.

**Comment by si458** [comment]
very helpful hint thanks!
you could add this info the help docs for other users!
submit a PR :)

**Comment by silversword411** [comment]
https://github.com/Ylianst/MeshCentral/commit/aca0e7aab4f660f4bb3ba626024227a116b326ff
